api/envoy/extensions/filters/http/admission_control/v3alpha/admission_control.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.admission_control.v3alpha;
+
+import "envoy/config/core/v3/base.proto";
+import "envoy/type/v3/range.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/wrappers.proto";
+import "google/rpc/status.proto";
+
+import "udpa/annotations/migrate.proto";
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.admission_control.v3alpha";
+option java_outer_classname = "AdmissionControlProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Admission Control]
+// [#extension: envoy.filters.http.admission_control]
+
+message AdmissionControl {
+  // Default method of specifying what constitutes a successful request. All status codes that
+  // indicate a successful request must be explicitly specified if not relying on the default
+  // values.
+  message SuccessCriteria {
+    message HttpCriteria {
+      // Status code ranges that constitute a successful request. Configurable codes are in the
+      // range [100, 600). If empty, all HTTP requests will be considered unsuccessful.
+      repeated type.v3.Int32Range http_success_status = 1;
+    }
+
+    message GrpcCriteria {
+      // Status codes that constitute a successful request. If empty, all gRPC requests will be
+      // considered unsuccessful.
+      // Mappings can be found at: https://github.com/grpc/grpc/blob/master/doc/statuscodes.md.
+      repeated uint32 grpc_success_status = 1;
+    }
+
+    // If HTTP criteria are unspecified, all HTTP status codes below 500 are treated as successful
+    // responses.
+    HttpCriteria http_criteria = 1;
+
+    // GRPC status codes to consider as request successes. If unspecified, defaults to: Ok,
```

Comment:
My reasoning for picking these specific codes is that something like `Unauthenticated` or `Unimplemented` or `InvalidArgument` won't be fixed by simply sending less requests. If a client starts sending bogus/malformed requests along with good ones, I don't want the admission controller to start rejecting the good requests if it won't make things better. In that contrived case, it would make things worse.

Does that change anything for you?

Reply:
I guess? I'm fine either way but I would flesh out the docs as to why the default is what it is. Same for the HTTP one.